packages/editor/src/components/collapsible-block-toolbar/index.js

```diff
@@ -9,29 +9,20 @@ import clsx from 'clsx';
 import {
 	BlockToolbar,
 	store as blockEditorStore,
-	privateApis as blockEditorPrivateApis,
 } from '@wordpress/block-editor';
 import { useEffect } from '@wordpress/element';
 import { Button, Popover } from '@wordpress/components';
 import { __ } from '@wordpress/i18n';
 import { next, previous } from '@wordpress/icons';
 import { useSelect } from '@wordpress/data';
 
-/**
- * Internal dependencies
- */
-import { unlock } from '../../lock-unlock';
-
-const { useHasBlockToolbar } = unlock( blockEditorPrivateApis );
-
-function CollapsableBlockToolbar( { isCollapsed, onToggle } ) {
+export default function CollapsibleBlockToolbar( { isCollapsed, onToggle } ) {
 	const { blockSelectionStart } = useSelect( ( select ) => {
 		return {
 			blockSelectionStart:
 				select( blockEditorStore ).getBlockSelectionStart(),
 		};
 	}, [] );
-	const hasBlockToolbar = useHasBlockToolbar();
```

Comment:
Removing this in favor of using it in the consuming component allows for better conditional rendering. Before the consumer can’t easily know if the there’s nothing rendered. The impetus is to be able to show the Document Bar when the block toolbar would be empty.

Reply:
Makes sense. Thanks for your comment. I guess the use cases for a way to check whether a block actually has toolbar controls are few so the effort to find a way of doing it that’s more performant has so far been spared. I like that you had some ideas https://github.com/WordPress/gutenberg/pull/58979#issuecomment-1971681017

> * Adding an `isBlockToolbarAvailable` state to the block editor store

It seems something like that would be good though I don’t know how it’d work.

Anyway, I've reverted this to have the PR focused on the layout improvement.